File: prototypes/Prototype_Mental_Health_Chat_Bot/prototype.js
const { HfInference } = require('@huggingface/inference');

const HF_TOKEN = 'insert_token_here'; // Replace with your actual token

const inference = new HfInference(HF_TOKEN);

const prompt = "What is mental health? How can I take care of my mental health?";

async function generateText(prompt) {
  const response = await inference.textGeneration({
    model: 'mistralai/Mistral-7B-v0.1',
    inputs: prompt,
    parameters: {
      max_new_tokens: 300,
      num_beams:10,
    no_repeat_ngram_size:10,
    early_stopping:true,
    //   do_sample: true,
    //   top_p: 0.5,
    //   temperature: 0.5,
    },
    options: {
      use_cache: false,
      wait_for_model: true,
    },
  });

//   console.log(response);
  return response.generated_text;
}


generateText(prompt)
  .then((result) => {
    console.log(result);
  })
  .catch((error) => {
    console.error(error);
  });

File: prototypes/Prototype_Mental_Health_Chat_Bot/prototype.py
# Source: https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.1-GGUF

from ctransformers import AutoModelForCausalLM

# Set gpu_layers to the number of layers to offload to GPU. Set to 0 if no GPU acceleration is available on your system.

llm = AutoModelForCausalLM.from_pretrained("TheBloke/Mistral-7B-Instruct-v0.1-GGUF", model_file="mistral-7b-instruct-v0.1.Q4_K_M.gguf", model_type="mistral", gpu_layers = 0)

print(llm("What is meantal health? How can i take care of my mental health?"))